src/core/load-fragment.c

```diff
@@ -776,8 +776,106 @@ int config_parse_socket_bindtodevice(
         return 0;
 }
 
-DEFINE_CONFIG_PARSE_ENUM(config_parse_output, exec_output, ExecOutput, "Failed to parse output specifier");
-DEFINE_CONFIG_PARSE_ENUM(config_parse_input, exec_input, ExecInput, "Failed to parse input specifier");
+DEFINE_CONFIG_PARSE_ENUM(config_parse_input, exec_input, ExecInput, "Failed to parse input literal specifier");
+DEFINE_CONFIG_PARSE_ENUM(config_parse_output, exec_output, ExecOutput, "Failed to parse output literal specifier");
+
+int config_parse_exec_input(const char *unit,
+                       const char *filename,
+                       unsigned line,
+                       const char *section,
+                       unsigned section_line,
+                       const char *lvalue,
+                       int ltype,
+                       const char *rvalue,
+                       void *data,
+                       void *userdata) {
+        ExecContext *c = data;
+        size_t len;
+
+        assert(data);
+        assert(filename);
+        assert(line);
+        assert(rvalue);
+
+        if (startswith(rvalue, "fd:") && (len = strlen(rvalue)) > 3) {
+                // Strip prefix and validate fd name
```

Comment:
Something is wrong with the parentheses here, `>` applies to the whole and-expression. I don't think you need to special-case an empty file descriptor name in this check. Also we prefer `/* */` comments (don't ask, I don't know why). So just do this:
```
      const char *name;

      name = startswith(rvalue, "fd:");
      if (name) {
                if (!fdname_is_valid(name)) {
                      ...
                }

                c->std_input = EXEC_INPUT_NAMED_FD;
                return free_and_strdup(&c->std_input_fdname, name);
      } else {
                ...
```
(free_and_strdup needs to be checked too :()